cmd/serve/s3/backend.go

```diff
@@ -213,6 +213,14 @@ func (b *s3Backend) GetObject(ctx context.Context, bucketName, objectName string
 	}, nil
 }
 
+// storeModtime sets both "mtime" and "X-Amz-Meta-Mtime" to val in b.meta.
+// Call this whenever modtime is updated.
+func (b *s3Backend) storeModtime(fp string, meta map[string]string, val string) {
+	meta["X-Amz-Meta-Mtime"] = val
+	meta["mtime"] = val
```

Comment:
It should only be necessary to set `X-Amz-Meta-Mtime` as the s3 backend turns it into the `mtime` metadata.

So I'm not 100% sure what this is needed for.

I think the `serve s3` code is a bit confused about exactly which parameter to use so we should probably unconfuse it rather than make it worse!

Reply:
Here's a test to reproduce:

```
go run ./fstest/test_all  -run '^TestBisyncRemoteRemote/ext_paths.*$' -timeout 30s -verbose -maxtries 1 -remotes TestS3Rclone:
```

If I remember correctly, I think the problem is that there are blocks like this which try `X-Amz-Meta-Mtime` first but fall back to using `mtime`:

https://github.com/rclone/rclone/blob/e9e9feb21eeb200782180e8532f2a0a95f745740/cmd/serve/s3/backend.go#L307-L321

And previously it was not always consistently saving the one it ended up using in memory.

And then there's also `Last-Modified` just to make things even more fun!

https://github.com/rclone/rclone/blob/e9e9feb21eeb200782180e8532f2a0a95f745740/cmd/serve/s3/backend.go#L194-L197

Reply:
I think we should probably fix `serve s3` to be consistent and use just one of these values, but I decided to cherry-pick your patch for v1.67 - thank you :-)